### PR TITLE
[wealth_dynamics] Update code and exercises

### LINKED ISSUE #321: [wealth_dynamics] Some suggestions on content and code
Hi @jstac

I was reading through the `wealth_dynamics` lecture and had a few small suggestions.

I’d be grateful if you could take a look when you have a chance.

### Opening paragraph

- [x] It might be helpful to add one sentence explaining why this lecture is a natural fit for JAX: the model involves simulating large cross sections of households, which suits JAX's array-based programming model and accelerator support.

### Wealth dynamics

- [x] In the savings rule explanation, `w \geq \bar w` should likely be `w \geq \hat w`, matching the threshold in the displayed savings function.

- [x] It may be helpful to clarify the timing convention: when updating from $w_t$ to $w_{t+1}$, the simulation draws $z_{t+1}$ and the period $t+1$ idiosyncratic shocks.

### Numba / JAX implementation

- [x] The simulation horizon and aggregate-state indexing seem inconsistent across the individual path, Numba cross-section, and JAX cross-section implementations.

  The model equation updates wealth from $w_t$ to $w_{t+1}$ using next-period shocks:

  $$w_{t+1} = (1 + r_{t+1}) s(w_t) + y_{t+1}$$

  The function `generate_aggregate_state_sequence(..., length=T)` returns `T + 1` aggregate states. It initializes `z[0]` at the stationary mean, then simulates `z[1], \ldots, z[T]`.

  This convention is used in the individual time-series simulation, where the update uses `z[t+1]`. In other words, `z[0]` is treated as the initial aggregate state, and the wealth updates use the simulated next-period aggregate states.

  The cross-section implementations seem to use a different convention. The Numba cross-section loop updates with `z[t]`, so the first update uses `z[0]` and the last simulated aggregate state is not used. The JAX cross-section implementations loop over the full `z_sequence`, so they also use `z[0]` and appear to apply one more update than the Numba cross-section code.

  It may be clearer to choose one timing convention and apply it consistently across the individual, Numba, and JAX simulations. For example, if `z[0]` is the initial aggregate state, then all wealth updates could use `z_sequence[1:]`, corresponding to the aggregate states used when moving from $w_t$ to $w_{t+1}$.

- [x] In the first JAX implementation, the PRNG key is used before splitting and the resulting `subkey` is unused. It may be clearer to use the standard JAX pattern: split first, then draw with the subkey.

### Exercises

- [x] In Exercises 2 and 3, `ax.plot(x, y, label='equality')` plots the last Lorenz curve again rather than the 45-degree equality line as shown below. This should likely be `ax.plot(x, x, label='equality')`.

Image

- [x] In Exercise 3, setting $\mu_r = -\sigma_r^2 / 2$ holds the mean of the lognormal idiosyncratic return component constant, not its variance. The parenthetical should likely say “mean” rather than “variance”.

- [x] In the exercise solutions, the same PRNG key is reused across parameter values. This is useful as a common-random-number comparison, but it may be worth stating explicitly so readers do not interpret it as accidental key reuse.

What do you think? Happy to submit a PR.

Best,
Longye

### PR DESCRIPTION
**Pull request by longye-tian**
Hi @jstac 

This PR update the lecture `wealth_dynamics` according to #321 .

In particular, this PR:

- Added a brief opening sentence explaining that large cross-section simulations make the lecture a natural fit for JAX.

- Fixed the savings-threshold notation so the text uses `\hat w`, matching the displayed savings rule.

- Added a timing note clarifying that period `t+1` shocks are drawn before updating wealth from `w_t` to `w_{t+1}`.

- Made Numba and JAX cross-section updates consistently skip `z_sequence[0]` and use `z_sequence[1:]`.

- Updated the first JAX loop to split the PRNG key before drawing random shocks.

- Fixed Exercises 2 and 3 so the equality benchmark is plotted as the 45-degree line.

- Corrected Exercise 3 to say `\mu_r = -\sigma_r^2 / 2` holds the lognormal mean constant, not the variance.

- Added a note explaining that reused PRNG seeds support common-random-number comparisons across parameter values.

### CODE REVIEW

File: lectures/wealth_dynamics.md [hunk 1]
```diff
@@ -68,6 +71,9 @@ Wealth evolves as follows:
     w_{t+1} = (1 + r_{t+1}) s(w_t) + y_{t+1}
 ```
 
+In the simulations below, $z_{t+1}$ and the period $t+1$ idiosyncratic shocks
```
Comment (jstac) [inline lectures/wealth_dynamics.md]: Actually I think we can skip this sentence @longye-tian .  (It makes the next one redundant and it's not really necessary.)

**Comment by jstac** [comment]
Many thanks @longye-tian .  One small change above.  Also, could you please update PRNGKey to key?

https://docs.jax.dev/en/latest/jax.random.html

(Perhaps we should do this globally in one pass?)

File: lectures/wealth_dynamics.md (hunk 1 above)
Comment (longye-tian) [reply lectures/wealth_dynamics.md]: Roger that 🫡

**Comment by longye-tian** [comment]
> Many thanks @longye-tian . One small change above. Also, could you please update PRNGKey to key?
> 
> https://docs.jax.dev/en/latest/jax.random.html
> 
> (Perhaps we should do this globally in one pass?)

Many thanks @jstac for your review, that sounds good to me. I updated this lecture first.

I use codex to do a quick scan. There are still 22 `PRNGKey` occurrences across the JAX lecture series, so I think a focused PR replacing those with `key` would be worthwhile. I/codex also noticed a few old `jax.readthedocs.io` links that could be updated to `docs.jax.dev`, perhaps as a separate small cleanup.

I’d be happy to put together a separate PR replacing `PRNGKey` with `key` across the JAX lecture series if you’d like.

**Comment by jstac** [comment]
Many thanks @longye-tian , I'm going to merge this.  Much appreciated.

Regarding the PRNGKey issue, it would be great if you could put in a PR.  Please also update the doc links, as you mentioned.  Perhaps the best thing to do is open an issue and ping @mmcky , so he knows what's going on --- since this will touch a lot of lectures.  He might have comments on how to proceed.

**Comment by mmcky** [comment]
thanks @jstac and @longye-tian
